torch_em/self_training/probabilistic_unet_trainer.py

```diff
@@ -0,0 +1,116 @@
+import time
+import torch
+import torch_em
+
+
+class DummyLoss(torch.nn.Module):
+    pass
+
+
+class ProbabilisticUNetTrainer(torch_em.trainer.DefaultTrainer):
+    """This trainer implements training for the 'Probabilistic UNet' of Kohl et al.: (https://arxiv.org/abs/1806.05034).
+    This approach combines the learnings from UNet and VAEs (Prior and Posterior networks) to obtain generative
+    segmentations. The heuristic trains by taking into account the feature maps from UNet and the samples from
+    the posterior distribution, estimating the loss and further sampling from the prior for validation.
+
+    Parameters:
+        clip_posterior [bool] - (default: False)
+        prior_samples [int] - (default: 16)
+        supervised_loss [callable] - (default: None)
+        supervised_loss_and_metric [callable] - (default: None)
+    """
+
+    def __init__(
+            self,
+            clip_posterior=False,
```

Comment:
I would suggest to just use a `clipping_value` parameter with default `None`. If set to `None` don't clip, otherwise use the `clipping_value`. (see also comment below).